components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/AbstractRequestBuilder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth2;
+
+import org.wso2.carbon.identity.base.IdentityException;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * This is a global level interface for building requests.
+ */
+public interface AbstractRequestBuilder {
+
+    /**
+     * Build the request.
+     *
+     * @param request Incoming HttpServletRequest.
+     * @return Built HttpServletRequest.
+     * @throws IdentityException IdentityException.
+     */
+    HttpServletRequest buildRequest(HttpServletRequest request) throws IdentityException;
+
+    /**
+     * Check whether the request builder can handle the incoming request.
+     *
+     * @param request Incoming HttpServletRequest.
+     * @return True if the request builder can handle the request.
+     */
+    boolean canHandle(HttpServletRequest request);
```

Comment:
Shall we remove the canHandle method and instead use a config to mention what request builder should be used to build the request. We can default to PAR request builder. Having a canHandler to determine handling can become messy as the params in the oauth request is usually the same. And also do we see a need to have multiple active request builders in the system?

Reply:
As per the online discussion, our implementation of PAR will be given a unique prefix for the request_uri `urn:ietf:params:oauth:par:request_uri:` and this will be used to check if this is a PAR request in the ParRequestBuilder.